pkg/systemd/services/service.jsx

```diff
@@ -40,147 +41,137 @@ function debug() {
-
-        if (this.state.unit_props === null)
-            return <EmptyStatePanel loading title={_("Loading...")} paragraph={this.props.unitId} />;
-
-        const serviceDetails = <ServiceDetails unit={this.state.unit_props}
-                                owner={this.props.owner}
-                                permitted={superuser.allowed}
-                                loadingUnits={this.state.reloading}
-                                isValid={this.props.unitIsValid}
-                                pinnedUnits={this.props.pinnedUnits}
-        />;
-
-        // resolve Alias name to primary ID
-        const cur_unit_id = this.state.unit_props.Id;
-
-        const unit_type = this.props.owner == "system" ? "UNIT" : "USER_UNIT";
-        const match = [
-            "_SYSTEMD_" + unit_type + "=" + cur_unit_id, "+",
-            "COREDUMP_" + unit_type + "=" + cur_unit_id, "+",
-            unit_type + "=" + cur_unit_id,
-        ];
-        const service_type = this.props.owner == "system" ? "service" : "user-service";
-        const url = "/system/logs/#/?prio=debug&" + service_type + "=" + cur_unit_id;
-
-        const load_state = this.state.unit_props.LoadState;
-
-        return (
-            <WithDialogs>
-                <Page groupProps={{ sticky: 'top' }}
-                      isBreadcrumbGrouped
-                      id="service-details"
-                      breadcrumb={
-                          <Breadcrumb>
-                              <BreadcrumbItem to={"#" + cockpit.location.href.replace(/\/[^?]*/, '')}>{_("Services")}</BreadcrumbItem>
-                              <BreadcrumbItem isActive>
-                                  {cur_unit_id}
-                              </BreadcrumbItem>
-                          </Breadcrumb>}>
-                    <PageSection>
-                        <Gallery hasGutter>
-                            <GalleryItem id="service-details-unit">{serviceDetails}</GalleryItem>
-                            {(load_state === "loaded" || load_state === "masked") &&
-                            <GalleryItem id="service-details-logs">
-                                <LogsPanel title={_("Service logs")} match={match} emptyMessage={_("No log entries")} max={10} goto_url={url} search_options={{ prio: "debug", [service_type]: cur_unit_id }} />
-                            </GalleryItem>}
-                        </Gallery>
-                    </PageSection>
-                </Page>
-            </WithDialogs>
-        );
-    }
-}
+    };
+
+    // load object path and set up PropertiesChanged subscription whenever unitId changes
+    useObject(
+        () => {
+            dbusClient.call(s_bus.O_MANAGER, s_bus.I_MANAGER, "LoadUnit", [unitId])
+                    .then(([path]) => {
+                        debug("Service detail", unitId, "loaded path", path);
+
+                        if (_path.current?.propSub)
```

Comment:
This is very nicely replace by opt chaining, _path.current?.propSub?.remove() and we can drop the condition.

Reply:
You are right! We don't even need the second ?, as propSub always gets set. Updated, thanks for pointing out!